graphql_compiler/schema_transformation/split_query.py

```diff
@@ -116,42 +214,67 @@ def split_query(query_ast, merged_schema_descriptor):
     return root_query_node, frozenset(name_assigner.intermediate_output_names)
 
 
-def _get_edge_to_stitch_fields(merged_schema_descriptor):
+def _get_edge_to_stitch_fields(
+    merged_schema_descriptor: MergedSchemaDescriptor,
+) -> Dict[Tuple[str, str], Tuple[str, str]]:
     """Get a map from type/field of each cross schema edge, to the fields that the edge stitches.
 
     This is necessary only because GraphQL currently doesn't process schema directives correctly.
     Once schema directives are correctly added to GraphQLSchema objects, this part may be
     removed as directives on a schema field can be directly accessed.
 
     Args:
-        merged_schema_descriptor: MergedSchemaDescriptor namedtuple, containing a schema AST
-                                  and a map from names of types to their schema ids
+        merged_schema_descriptor: containing a schema AST and a map from names of types to
+                                  their schema ids.
 
     Returns:
-        Dict[Tuple(str, str), Tuple(str, str)], mapping (type name, vertex field name) to
-        (source field name, sink field name) used in the @stitch directive, for each cross
-        schema edge
+        Dictionary mapping (type name, vertex field name) to (source field name, sink field name)
+        used in the @stitch directive, for each cross schema edge.
     """
     edge_to_stitch_fields = {}
     for type_definition in merged_schema_descriptor.schema_ast.definitions:
         if isinstance(type_definition, (ObjectTypeDefinitionNode, InterfaceTypeDefinitionNode)):
             for field_definition in type_definition.fields:
                 stitch_directive = try_get_ast_by_name_and_type(
-                    field_definition.directives, "stitch", DirectiveNode
+                    list(field_definition.directives), "stitch", DirectiveNode
```

Comment:
I added this `list()` here and in other places to get rid of mypy errors like `Argument 1 to "try_get_ast_by_name_and_type" has incompatible type "List[FieldNode]"; expected "Optional[List[Node]]"`. Not sure if this is the best way to go about it. I used these docs: https://mypy.readthedocs.io/en/stable/common_issues.html#invariance-vs-covariance. In some cases, I tried adding explicit type annotations to variables, but was still seeing errors so tried this method instead. Let me know if there's a better way!

Reply:
Would it be feasible to change the type hint of `try_get_ast_by_name_and_type` from `Optional[List[Node]]` to `Optional[Sequence[Node]]` which I believe will be covariant? I think that function doesn't actually modify the list, so it shouldn't need to know that the type is a list and not a sequence.

Using abstract `Sequence/AbstractSet/Mapping` instead of concrete `List/Set/Dict` when possible is something we should probably try to get better at over time.